### ISSUE TITLE
.NET SDK 5:  Could not load file or assembly Microsoft.CodeAnalysis

### ISSUE THREAD

**Issue by kevinoid** [dotnet/roslynator#878]
Creating a new issue similar (identical?) to #853, which was closed.

**Product and Version Used**:  `Roslynator.Analyzers` 4.0.0 and .NET SDK 5.0.404

**Steps to Reproduce**: Run the following:

```sh
dotnet new console
dotnet add package Roslynator.Analyzers --version 4.0.0
dotnet build
```

**Actual Behavior**:  ~350 CS8032 warnings such as:

> CSC : warning CS8032: An instance of analyzer Roslynator.CSharp.Analysis.AbstractTypeShouldNotHavePublicConstructorsAnalyzer cannot be created from /home/username/.nuget/packages/roslynator.analyzers/4.0.0/analyzers/dotnet/cs/Roslynator.CSharp.Analyzers.dll : Could not load file or assembly 'Microsoft.CodeAnalysis, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.. [/path/to/csproj]

**Expected Behavior**: No CS8032 warnings.

**Comment by JosefPihrt** [comment]
Hi,

Thanks for creating new issue.

Roslynator 4.0 requires Roslyn 4.0.1 (Microsoft.CodeAnalysis.dll and others).

To use Roslynator analyzers you need one of following tools:

* VS 2022 (for diagnostics inside IDE)
* .NET SDK 6.0  (for diagnostics at command line)

**Comment by JosefPihrt** [comment]
If you cannot use .NET SDK 6 for some reason you can use Roslynator.Analyzers 3.3.0

**Comment by kevinoid** [comment]
Thanks @JosefPihrt, that's good to know.  My apologies if I missed that in the documentation.  I don't suppose there's a way to declare that in the package metadata to give users a more comprehensible error message?  If there's nothing to fix here, feel free to close.

**Comment by marcospgp** [comment]
So having updated the nuget packages might be why all Roslynator analyzers stopped working for me? Unity is on .NET 4.x, so I'm unable to use the latest release with it?

**Comment by sharwell** [comment]
> Unity is on .NET 4.x, so I'm unable to use the latest release with it?

Unity doesn't support Roslyn 4.x, but that's completely unrelated to the use of .NET Framework. Roslyn still ships net472 binaries with every build. This restriction is partially mentioned here in the Source Generators section:
https://docs.unity3d.com/2022.2/Documentation/Manual/roslyn-analyzers.html

**Comment by marcospgp** [comment]
@sharwell I can't find a reference to Roslyn 4.x support on that Unity docs link, maybe this has since been updated?

**Comment by sharwell** [comment]
@marcospgp It's mentioned indirectly:

> Install the Microsoft.CodeAnalysis NuGet package. Your source generator must use [Microsoft.CodeAnalysis 3.8](https://www.nuget.org/packages/Microsoft.CodeAnalysis.CSharp/3.8.0) to work with Unity.

Source generators (`ISourceGenerator` and later `IIncrementalGenerator`) and analyzers (`DiagnosticAnalyzer`) are defined in the same assembly, so a version restriction for one implies the same version restriction for the other.

**Comment by josefpihrt** [comment]
Roslynator analyzers now supports Roslyn 3.8 (see #1349)
